src/test/java/org/codehaus/plexus/archiver/zip/ArchiveFileComparator.java

```diff
@@ -1,81 +1,76 @@
 package org.codehaus.plexus.archiver.zip;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.UncheckedIOException;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
 import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
 import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
 import org.apache.commons.compress.archivers.zip.ZipFile;
-import org.codehaus.plexus.archiver.ArchiveFile;
+import org.codehaus.plexus.archiver.tar.TarFile;
 import org.codehaus.plexus.util.IOUtil;
-import org.junit.Assert;
 
 /**
  * A utility class, which allows to compare archive files.
  */
 public class ArchiveFileComparator
 {
 
-    public interface TarArchiveEntryConsumer
-    {
-
-        void accept( TarArchiveEntry entry ) throws IOException;
-
-    }
-
-    public static void forEachTarArchiveEntry( ArchiveFile file, TarArchiveEntryConsumer consumer )
-        throws IOException
+    public static void forEachTarArchiveEntry( TarFile file, Consumer<TarArchiveEntry> consumer )
     {
-        for ( java.util.Enumeration en = file.getEntries(); en.hasMoreElements(); )
-        {
-            TarArchiveEntry ze = (TarArchiveEntry) en.nextElement();
-            if ( !ze.isDirectory() )
-            {
-                consumer.accept( ze );
-            }
-        }
+        tarEntries( file )
```

Comment:
Looks like other usages of `tarEntries` also skip the directories. Maybe we can rename it `tarFileEntries` or something like that and move the `!entry.isDirectory()` filter there to avoid code duplication.